fs/config/rc_test.go

```diff
@@ -116,6 +116,47 @@ func TestRc(t *testing.T) {
 		assert.Equal(t, "cabbage", obscure.MustReveal(FileGet(testName, "test_key2")))
 	})
 
+	t.Run("Upload", func(t *testing.T) {
+		// Call upload with implicit nosave == false
+		call := rc.Calls.Get("config/upload")
+		assert.NotNil(t, call)
+		in := rc.Params{
+			"content": "[" + testName + "]\n" +
+				"type = local\n" +
+				"test_key = rutabaga\n" +
+				"test_key2 = cabbage\n",
+		}
+		out, err := call.Fn(context.Background(), in)
+		require.NoError(t, err)
+		assert.Nil(t, out)
+
+		assert.Equal(t, "local", FileGet(testName, "type"))
```

Comment:
@ncw I had to remove `MustReveal` from the asserts which is present in other lines. Is this a problem?

Reply:
No that is fine I think - passwords don't get revealed until they are used - they are kept encrypted in the config file